File: Chapter10/Q2.java
import java.util.*;

/**
 * Q2
 */
public class Q2 {

    public static void main(String[] args) {
        String[] test = {"eat","tea","tan","ate","nat","bat"};
        for(String ele : Sort(test))
            System.out.println(ele);
    }

    public static String[] Sort(String[] strs) {
        Arrays.sort(strs,new Comparator<String>(){
            public int compare(String a, String b){
                char[] A = a.toCharArray();
                char[] B =b.toCharArray();
                Arrays.sort(A);
                Arrays.sort(B);
                return new String(A).compareTo(new String(B));
            }
        });

        return strs;
    }
}

File: Chapter10/Q2.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        dic = {}
        res = []
        for str in strs:
            temp = ''.join(sorted(str))
            if temp in dic:
                dic[temp].append(str)
            else:
                dic[temp] = [str]
                res.append(dic[temp])
        return res

test = Solution()

print test.groupAnagrams(["eat","tea","tan","ate","nat","bat"])
